src/main/java/com/example/demo/controller/MailSenderController.java

```diff
@@ -0,0 +1,44 @@
+package com.example.demo.controller;
+
+import com.example.demo.model.Message;
+import com.example.demo.repo.JsonRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController()
+@RequestMapping("/message")
+public class MailSenderController {
+
+    @Autowired
+    private JsonRepo jsonRepo;
+
+    @GetMapping()
+    public List<Message> getMessages() {
+        List<Message> sender = jsonRepo.loadFromJsonToList();
+        return sender;
+    }
+
+    @GetMapping("/{id}")
+    public Message getMessage(@PathVariable int id) {
+        Message message = jsonRepo.findById((long) id);
+        return message;
+    }
+
+    @PostMapping("/new")
+    public void addMessage(@RequestBody Message message) {
+        jsonRepo.addMessage(message);
+    }
+
+    @PutMapping("/updateDate/{id}/{date}")
```

Comment:
Again, you are trying to give a name. Let's try to break an access into steps. First you are looking on collection of **/messages**. You are interested in specific message identified by an **id**. If we join all parts in bold we will get URL identifier for specific message: **/messages/{id}**. You want to update an information that associated with this message or *PUT* new future date. So the mapping will be:
`@PutMapping('/{id}')`. The same as you have for get message! And it's perfectly fine, cause you are dealing with the same resource, but you're using different http methods to tell what you want to do with it.
Then a new value for the date shouldn't be inside URL. You should send it in request body. The simplest way to do it will be using form-urlencoded:
`
@PutMapping('/{id}')
public void updateFutureDate(@PathVariable long id, @PleaseFindAnnotationToExtractParameterFromBody long date)
`

Reply:
Done